.github/workflows/rollback.yml

```diff
@@ -1,51 +1,93 @@
-name: Rollback release
+name: Rollback latest release
 
 on:
   workflow_dispatch:
     inputs:
-      version-to-roll-back:
-        description: version number that needs to be rolled back (format x.x.x)
-        type: string
+      rollbackChannel:
+        description: Which channel needs to be rolled back?
+        type: choice
         required: true
+        options:
+        - stable
+        - beta
+        - alpha
 
 jobs:
-  determine-target-tag:
-    name: Determine target tag for rollback
+  get-target-and-rollback-tags:
+    name: Get target tag and rollback tags
     runs-on: ubuntu-latest
     outputs:
-      targetTag: ${{ github.ref_name }}
-      isStableRelease: ${{ steps.isStableRelease.outputs.isStableRelease }}
+      targetTag: ${{ steps.getRollbackTargetTags.outputs.targetTag }}
+      rollbackTag: ${{ steps.getRollbackTargetTags.outputs.rollbackTag }}
     steps:
-      - name: is input a tag
-        if: github.ref_type != 'tag'
-        run: exit 1
-      - name: Get target version
-        id: isStableRelease
-        shell: bash
+      - name: install apt-get dependencies
         run: |
-          if [[ ${{ github.ref_name }} =~"beta" ]]
-          then
-            echo "isStableRelease=true" >> $GITHUB_OUTPUT
+          sudo apt-get update
+          sudo apt-get install -y awscli jq
+      - name: get rollback and target tags
+        id: getRollbackTargetTags
+        run: |
+          VERSION_LIST=$(npm view heroku versions --json)
+          CHANNEL_VERSION_LIST=[]
+          if [[ ${{ inputs.rollbackChannel }} == 'stable' ]]; then
+            CHANNEL_VERSION_LIST=$(echo ${VERSION_LIST} | jq '[.[] | select(test("-") | not )]')
           else
-            echo "isStableRelease=false" >> $GITHUB_OUTPUT
+            CHANNEL_VERSION_LIST=$(echo ${VERSION_LIST} | jq '[.[] | select(test("'${{ inputs.rollbackChannel }}'"))]')
           fi
+          ROLLBACK_VERSION=$(echo ${CHANNEL_VERSION_LIST} | jq '.[-1]')
+          echo "rollbackTag=${ROLLBACK_VERSION}" >> $GITHUB_OUTPUT
+          TARGET_VERSION=$(echo ${CHANNEL_VERSION_LIST} | jq '.[-2]')
+          echo "targetTag=${TARGET_VERSION}" >> $GITHUB_OUTPUT
```

Comment:
Final sanity check: how worried are we that we can not specify the version to rollback to?